openbsd/Platform.c

```diff
@@ -375,3 +376,49 @@ void Platform_getBattery(double* percent, ACPresence* isOnAC) {
          *isOnAC = s.value;
    }
 }
+
+char** Platform_getLocalIPv4addressChoices(Meter* meter) {
+   // TODO
+   (void) meter;
+   return NULL;
+}
+
+char** Platform_getLocalIPv6addressChoices(Meter* meter) {
+   // TODO
+   (void) meter;
+   return NULL;
+}
+
+void Platform_getLocalIPv4address(const char* choice, char* buffer, size_t size) {
+   // TODO
+   (void) choice;
+   (void) buffer;
+   (void) size;
+   return;
+}
+
+void Platform_getLocalIPv6address(const char* choice, char* buffer, size_t size) {
+   // TODO
+   (void) choice;
+   (void) buffer;
+   (void) size;
+   return;
+}
+
+char **Platform_getDiskUsageChoices(Meter* meter) {
+   // TODO
+   (void) meter;
+   return NULL;
+}
+
+void Platform_getDiskUsage(const char* choice, DiskUsageData *data) {
+   // TODO
+   (void) choice;
+   (void) data;
+}
+
+char **Platform_getDynamicMeterChoices(Meter* meter) {
+   // TODO
+   (void) meter;
+   return NULL;
+}
```

Comment:
Still need to do something for each of these.  It makes me wonder though, for PCP we have all of the component instance names readily available to us for a meter - could we make a generic implementation of getChoices() for PCP and have every meter that uses metrics with an instance domain be one of these 'choice' meters, automatically?

Reply:
> could we make a generic implementation of getChoices() for PCP and have every meter that uses metrics with an instance domain be one of these 'choice' meters, automatically?

Sorry, I don't understand this part! I have a generic function that returns all the choices here:
(that is actually return the external instance name list)
https://github.com/htop-dev/htop/blob/7d6d345e286fe890cd504be7522a02ed76bee8c8/DynamicMeter.c#L132
https://github.com/htop-dev/htop/blob/7d6d345e286fe890cd504be7522a02ed76bee8c8/pcp/Platform.c#L869
https://github.com/htop-dev/htop/blob/7d6d345e286fe890cd504be7522a02ed76bee8c8/pcp/PCPDynamicMeter.c#L480

Reply:
and the `curChoice` got saved in htoprc...

Reply:
@smalinux aha, right you are - so the remaining piece then is those TODO items in the code relating to specific meters like network/hostname ... can we just call Platform_getDynamicMeterChoices there too?

Reply:
@natoscott exactly :)

Reply:
> ... can we just call Platform_getDynamicMeterChoices there too?

I do not think so! these Meters (network/hostname) depend on these Metrics: https://github.com/htop-dev/htop/pull/200#discussion_r907890905
I think I should provide a static implementation for them like any normal Meter... like this: #992

or we could ignore these TODO items completely and provide the same Meters as DynamicMeters via pcp/meters/ config files...